### PR TITLE
fix - useOverlayPosition mount flicker

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
Closes https://github.com/adobe/react-spectrum/issues/1694

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/1694).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:
One can check this [storybook example](http://localhost:9003/?path=/story/useoverlayposition--positioned-container-bottom). 
Sometimes after reload clicking on trigger leads to flicker. Note - Only happens on first mount as new position gets stored and it's used in re-renders.

## 🧢 Your Project:
GeekyAnts

### CODE REVIEW

File: packages/@react-aria/overlays/src/useOverlayPosition.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@
  */
 
 import {calculatePosition, PositionResult} from './calculatePosition';
-import {HTMLAttributes, RefObject, useCallback, useEffect, useRef, useState} from 'react';
+import {HTMLAttributes, RefObject, useCallback, useLayoutEffect, useRef, useState} from 'react';
```
Comment (LFDanLu) [inline packages/@react-aria/overlays/src/useOverlayPosition.ts:14]: Looks like the SSR tests are failing, think you can use `useLayoutEffect` from the react-aria package to fix

**Review by LFDanLu — CHANGES_REQUESTED** [review]

**Review by LFDanLu — COMMENTED** [review]
Looks like this breaks `MenuTrigger` menu rendering. I was able to fix it locally by updating `MenuTrigger` so that it has a similar useLayoutEffect as Picker, but I need to dig some more to see if this is the appropriate fix

**Comment by LFDanLu** [comment]
@intergalacticspacehighway Just an update, got a little bit of time to look and a possible alternate fix to the MenuTrigger problem is to change the following useEffects to useLayoutEffects: 
https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/usePress.ts#L75
https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/menu/src/Menu.tsx#L37-L44

Now that useOverlayPosition uses `useLayoutEffect`, we need to sync the refs that get provided by the context providers during the same phase otherwise useOverlayPosition can't calculate the proper position. Haven't tested this too thoroughly, not sure if it would break anything else, will hopefully get some more time next week to look

**Comment by LFDanLu** [comment]
@intergalacticspacehighway Sorry about the delay, got some time today to test the changes I mentioned before. It doesn't look like they break anything, gonna push those changes to your branch and get another set of eyes to take a look.

**Review by snowystinger — APPROVED** [review]
I tested in the useOverlayPosition stories, it has gotten rid of the flicker. I'm not seeing any issues with autofocus, grid focus movement, or menus (quick things to check). We'll have to make a good pass of this in our testing session, the change is subtle.

**Review by devongovett — APPROVED** [review]

**Pull request merged**